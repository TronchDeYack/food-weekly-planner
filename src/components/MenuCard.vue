<template>
  <v-card class="mt-3 mb-3">
    <v-card-title>{{title}}</v-card-title>
    <v-card-text>
      <v-row align="center">
        <v-col class="d-flex align-center">
          <Timer />
          {{minutes}} min
        </v-col>
        <v-col>
          <Calories :class="caloriesIndicator"/>
        </v-col>
      </v-row>
    </v-card-text>
    <v-divider class="mx-4"></v-divider>
    <v-card-text>
      Contient
      <v-chip-group column>
        <v-chip label><Egg :disabled="!hasEgg"/></v-chip>
        <v-chip label><Fish :disabled="!hasFish"/></v-chip>
        <v-chip label><Meat :disabled="!hasMeat"/></v-chip>
        <v-chip label><Vegetable :disabled="!isVegetarian"/></v-chip>
      </v-chip-group>
    </v-card-text>
  </v-card>
</template>

<style lang="scss">
svg {
  width: 30px;
  height: 30px;
}
</style>

<script>
import Calories from '@/components/icons/Calories.vue';
import Egg from '@/components/icons/Egg.vue';
import Fish from '@/components/icons/Fish.vue';
import Meat from '@/components/icons/Meat.vue';
import Timer from '@/components/icons/Timer.vue';
import Vegetable from '@/components/icons/Vegetable.vue';

export default {
  props: {
    title: {
      type: String,
    },
    minutes: {
      type: Number,
    },
    caloriesIndicator: {
      type: String,
    },
    hasEgg: {
      type: Boolean,
    },
    hasMeat: {
      type: Boolean,
    },
    hasFish: {
      type: Boolean,
    },
    isVegetarian: {
      type: Boolean,
    },
  },
  components: {
    Calories,
    Egg,
    Fish,
    Meat,
    Timer,
    Vegetable,
  },
};
</script>
