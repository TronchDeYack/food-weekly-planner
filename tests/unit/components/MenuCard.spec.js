import { shallowMount } from '@vue/test-utils';

import MenuCard from '@/components/MenuCard.vue';

describe('MenuCard', () => {
  const ingredients = [
    {
      id: 'PATE-ID',
      label: 'PÃ¢tes',
      quantity: 200,
      unit: 'gr',
      type: 'STARCHY-FOOD',
    },
    {
      id: 'SAUCE-TOMATE-ID',
      label: 'Sauce Tomate',
      quantity: 200,
      unit: 'GR',
      type: 'CONDIMENT',
    },
  ];

  const propsData = {
    id: 'ID',
    title: 'TITLE',
    minutes: 30,
    caloriesLevel: 'high',
    ingredients,
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(MenuCard, { propsData });
  });

  it('Should render.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('getRecipeDetailsUrl', () => {
    it('Should redirect to recipe details when clicking on the card.', () => {
      expect(wrapper.find('v-card-stub').attributes('to')).toEqual(`/recipes/${propsData.id}`);
    });
  });

  describe('computed', () => {
    const ingredientsWithMeat = [
      ...ingredients,
      {
        id: 'STEAK-ID',
        label: 'Steak',
        quantity: 100,
        unit: 'GR',
        type: 'MEAT',
      },
    ];
    const propsDataWithMeat = {
      ...propsData,
      ingredients: ingredientsWithMeat,
    };

    const ingredientsWithFish = [
      ...ingredients,
      {
        id: 'SALMON-ID',
        label: 'Saumon',
        quantity: 100,
        unit: 'GR',
        type: 'FISH',
      },
    ];
    const propsDataWithFish = {
      ...propsData,
      ingredients: ingredientsWithFish,
    };

    describe('hasEgg', () => {
      it('Should highlight "egg" icon when there is an egg in the ingredients.', () => {
        const ingredientsWithEgg = [
          ...ingredients,
          {
            id: 'EGG-ID',
            label: 'Oeuf',
            quantity: 2,
            unit: 'unit',
            type: 'EGG',
          },
        ];
        const propsDataWithEgg = {
          ...propsData,
          ingredients: ingredientsWithEgg,
        };

        wrapper = shallowMount(MenuCard, { propsData: propsDataWithEgg });

        expect(wrapper.vm.hasEgg).toBe(true);
      });

      it('Should not highlight "egg" icon when there is no egg in the ingredients.', () => {
        expect(wrapper.vm.hasEgg).toBe(false);
      });
    });

    describe('hasMeat', () => {
      it('Should highlight "meat" icon when there is meat in the ingredients.', () => {
        wrapper = shallowMount(MenuCard, { propsData: propsDataWithMeat });

        expect(wrapper.vm.hasMeat).toBe(true);
      });

      it('Should not highlight "meat" icon when there is no meat in the ingredients.', () => {
        expect(wrapper.vm.hasMeat).toBe(false);
      });
    });

    describe('hasFish', () => {
      it('Should highlight "fish" icon when there is fish in the ingredients.', () => {
        wrapper = shallowMount(MenuCard, { propsData: propsDataWithFish });

        expect(wrapper.vm.hasFish).toBe(true);
      });

      it('Should not highlight "fish" icon when there is no fish in the ingredients.', () => {
        expect(wrapper.vm.hasFish).toBe(false);
      });
    });

    describe('isVegetarian', () => {
      it('Should highlight "vegetarian" icon when there is neither meat nor fish in the ingredients.', () => {
        expect(wrapper.vm.isVegetarian).toBe(true);
      });

      it('Should not highlight "vegetarian" icon when there is fish in the ingredients.', () => {
        wrapper = shallowMount(MenuCard, { propsData: propsDataWithFish });

        expect(wrapper.vm.isVegetarian).toBe(false);
      });

      it('Should not highlight "vegetarian" icon when there is meat in the ingredients.', () => {
        wrapper = shallowMount(MenuCard, { propsData: propsDataWithMeat });

        expect(wrapper.vm.isVegetarian).toBe(false);
      });
    });
  });
});
