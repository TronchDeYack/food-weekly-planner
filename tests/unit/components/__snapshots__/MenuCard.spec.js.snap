// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCard Should render. 1`] = `
<v-card-stub loaderheight="4" to="/recipes/ID" tag="div" class="mt-3 mb-3">
  <v-card-title-stub>TITLE</v-card-title-stub>
  <v-card-text-stub>
    <v-row-stub tag="div" align="center">
      <v-col-stub tag="div" class="d-flex align-center">
        <timer-stub></timer-stub>
        30 recipes.minutes
      </v-col-stub>
      <v-col-stub tag="div">
        <calories-stub class="high"></calories-stub>
      </v-col-stub>
    </v-row-stub>
  </v-card-text-stub>
  <v-divider-stub class="mx-4"></v-divider-stub>
  <v-card-text-stub>
    recipes.contains
    <v-chip-group-stub activeclass="v-slide-item--active" nexticon="$next" mobilebreakpoint="1264" previcon="$prev" column="true">
      <v-chip-stub activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete" label="true">
        <egg-stub disabled="true"></egg-stub>
      </v-chip-stub>
      <v-chip-stub activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete" label="true">
        <fish-stub disabled="true"></fish-stub>
      </v-chip-stub>
      <v-chip-stub activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete" label="true">
        <meat-stub disabled="true"></meat-stub>
      </v-chip-stub>
      <v-chip-stub activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete" label="true">
        <vegetable-stub></vegetable-stub>
      </v-chip-stub>
    </v-chip-group-stub>
  </v-card-text-stub>
</v-card-stub>
`;
